vision/standard_object/odlc_classify_shape.py

```diff
@@ -69,420 +95,283 @@ def process_shapes(
+    x, y = condense_polar(pol_cnt_sorted)
 
-def get_angles(approx: consts.Contour) -> npt.NDArray[npt.Shape["*"], npt.Float64]:
-    """
-    Will find the angle at every point in the given approximated contour.
+    return y
 
-    Parameters
-    ----------
-    approx : consts.Contour
-        Approximated version of the contour (polygon approximation with cv2.approxPolyDP())
 
-    Returns
-    -------
-    angles : npt.NDArray[npt.Shape["*"], npt.Float64]
-        1d array of angles for each corresponding vertex in the given approximated contour
+def condense_polar(
+    polar_array: NDArray[Shape["*, 2"], Float64]
+) -> NDArray[Shape["128,2"], Float64]:
     """
-    angles: npt.NDArray[npt.Shape["*"], npt.Float64] = np.empty(approx.shape[0], npt.Float64)
-    idx: int
-    a_pt: npt.NDArray[npt.Shape["1, 2"], npt.IntC]  # one point of the approx contour
-    for idx, a_pt in enumerate(approx):
-        angles[idx] = get_angle((approx[idx - 1], a_pt, approx[(idx + 1) % approx.shape[0]]))
-
-    return angles
-
-
-def compare_angles(
-    angles: npt.NDArray[npt.Shape["*"], npt.Float64], compare_angle: float, thresh: float
-) -> npt.NDArray[npt.Shape["*"], npt.Bool8]:
-    """
-    Takes an array, angles, and compares each element against a specific angle.
+    Condenses a polar array to have 'NUM_STEPS' data points for analysis
 
     Parameters
     ----------
-    angles : npt.NDArray[npt.Shape["*"], npt.Float64]
-        An array of angles of some contour (assumed in degrees but would work if all values are
-        the same unit)
-    compare_angle : float
-        The value to compare each element of angles to.
-    thresh : float
-        The range on each side of angle that each element of angles may be and still pass.
+    polar_array : NDArray[Shape["*, 2"], Float64]
+        An array of polar points of unknown length
 
     Returns
     -------
-    are_valid : npt.NDArray[npt.Shape["*"], npt.Bool8]
-        Each element represents whether the corresponding element of angles was within thresh
-        of angle
-    """
-    are_valid: npt.NDArray[npt.Shape["*"], npt.Bool8] = np.empty(angles.shape[0], npt.Bool8)
-    idx: int
-    ang: npt.Float64  # each individual angle in angles
-    for idx, ang in enumerate(angles):
-        are_valid[idx] = abs(ang - compare_angle) < thresh
+    new_x, new_y : PolarArray, PolarArray
 
-    return are_valid
+    """
+    # Converting data to a form able to be passed into scipy.interp1d
+    x: NDArray[Shape["*"]] = np.empty(len(polar_array))
+    y: NDArray[Shape["*"]] = np.empty(len(polar_array))
+    for i in range(len(polar_array)):
+        x[i] = polar_array[i][1]
+        y[i] = polar_array[i][0]
+
+    # Linear interpolation to normalize all shapes to have the same number of data points
+    # Create the scipy interpolation model and initializes the x-values to be uniformly spaced
+    new_x: PolarArray = np.linspace(x.min(), x.max(), num=NUM_STEPS)
```

Comment:
`x.min()` and `x.max()` should always have the same values, 0 to 2*pi
```suggestion
    new_x: PolarArray = np.linspace(0, 2 * np.pi, num=NUM_STEPS)
```